sources/packages/backend/libs/integrations/src/esdc-integration/student-loan-balances/student-loan-balances.processing.service.ts

```diff
@@ -0,0 +1,133 @@
+import { Injectable } from "@nestjs/common";
+import { ConfigService, ESDCIntegrationConfig } from "@sims/utilities/config";
+import { ProcessSFTPResponseResult } from "../models/esdc-integration.model";
+import { StudentLoanBalancesIntegrationService } from "./student-loan-balances.integration.service";
+import { StudentLoanBalancesSFTPResponseFile } from "./models/student-loan-balances.model";
+import {
+  StudentLoanBalancesService,
+  StudentService,
+} from "@sims/integrations/services";
+import * as path from "path";
+import { InjectLogger, LoggerService } from "@sims/utilities/logger";
+import { Student } from "@sims/sims-db";
+import { StudentLoanBalances } from "@sims/sims-db/entities/student-loan-balances.model";
+import { getISODateOnlyString } from "@sims/utilities";
+
+/**
+ * Manages to process the Student Loan Balances files
+ * once they are available in the SFTP location.
+ */
+@Injectable()
+export class StudentLoanBalancesProcessingService {
+  private readonly esdcConfig: ESDCIntegrationConfig;
+  constructor(
+    config: ConfigService,
+    private readonly studentLoanBalancesIntegrationService: StudentLoanBalancesIntegrationService,
+    private readonly studentLoanBalancesService: StudentLoanBalancesService,
+    private readonly studentService: StudentService,
+  ) {
+    this.esdcConfig = config.esdcIntegration;
+  }
+
+  /**
+   * Download all files from SFTP and process them all.
+   * @returns Summary with what was processed and the list of all errors, if any.
+   */
+  async processStudentLoanBalances(): Promise<ProcessSFTPResponseResult[]> {
```

Comment:
I was expecting that the methods would be built based on the most updated libs. For instance, the log would be captured using the `ProcessSummary` as it was in the most recent schedulers.